nova_plugin/server.py

```diff
@@ -228,41 +226,22 @@ def rename(name):
     ctx.logger.debug(
         "server.create() server after transformations: {0}".format(server))
 
-    # First parameter is 'self', skipping
-    params_names = inspect.getargspec(nova_client.servers.create).args[1:]
-
-    params_default_values = inspect.getargspec(
-        nova_client.servers.create).defaults
-    params = dict(itertools.izip(params_names, params_default_values))
-
-    # Fail on unsupported parameters
-    for k in server:
-        if k not in params:
-            raise NonRecoverableError(
-                "Parameter with name '{0}' must not be passed to"
-                " openstack provisioner (under host's "
-                "properties.nova.instance)".format(k))
-
-    for k in params:
-        if k in server:
-            params[k] = server[k]
-
-    if not params['meta']:
-        params['meta'] = dict({})
+    if 'meta' not in server:
+        server['meta'] = dict({})
```

Comment:
just use dict() instead of dict({})
(yes i know its been there before as well :))